codeSnippets/snippets/tutorial-full-stack-task-manager/composeApp/src/commonMain/kotlin/org/example/ktor/network/httpClient.kt

```diff
@@ -0,0 +1,22 @@
+package org.example.ktor.network
+
+import io.ktor.client.HttpClient
+import io.ktor.client.plugins.contentnegotiation.ContentNegotiation
+import io.ktor.client.plugins.defaultRequest
+import io.ktor.serialization.kotlinx.json.json
+import kotlinx.serialization.json.Json
+
+val httpClient = HttpClient {
+    install(ContentNegotiation) {
+        json(Json {
+            encodeDefaults = true
+            isLenient = true
+            coerceInputValues = true
+            ignoreUnknownKeys = true
+        })
+    }
+    defaultRequest {
+        host = "1.2.3.4"
+        port = 8080
+    }
+}
```

Comment:
You may want to introduce an extension function here instead.  This way you can avoid the static instantiation and reuse the configuration in your test code.

Something like:

```kotlin
fun HttpClient.configuredForApi() = config {
    install(ContentNegotiation) {
        json(Json {
            encodeDefaults = true
            isLenient = true
            coerceInputValues = true
            ignoreUnknownKeys = true
        })
    }
    defaultRequest {
        host = "1.2.3.4"
        port = 8080
    }
}
```

Reply:
Thank you for the suggestion! I've updated the code to use an extension function as you recommended. By introducing the configuredForApi extension function, the configuration logic is now reusable across both the main application code 😄

However, in the test code, I utilized testApplication, which does not require the use of the defaultRequest block. Therefore, I created a separate extension function specifically tailored for the test environment.